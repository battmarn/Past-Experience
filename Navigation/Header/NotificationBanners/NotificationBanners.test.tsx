import React from 'react';
import * as reactRedux from 'react-redux';

import { renderWithStore } from 'test/helpers/renderWithStore';

import { NotificationBanners } from './index';

jest.mock(
  'react-redux',
  () =>
    ({
      ...jest.requireActual('react-redux'),
    } as jest.Mock),
);

const mockNotifications = [
  {
    text: 'Wrong email or password, please try again.',
    type: 'error',
    scroll: false,
    persistOnLocationChange: true,
  },
];

describe('<NotificationBanners />', () => {
  beforeEach(() => {
    jest.spyOn(reactRedux, 'useSelector').mockReturnValue(mockNotifications);
  });

  it('will render notification banner', () => {
    expect(renderWithStore(<NotificationBanners isNavbarMinimised={false} />).container).toMatchSnapshot();
  });

  it('will render notification banner with minimised class', () => {
    expect(renderWithStore(<NotificationBanners isNavbarMinimised />).container).toMatchSnapshot();
  });
});
