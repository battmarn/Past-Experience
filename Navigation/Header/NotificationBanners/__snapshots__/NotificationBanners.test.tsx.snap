// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBanners /> will render notification banner 1`] = `
<div>
  <div
    class="banners"
  >
    <div
      class="root error"
    >
      <p
        class="paragraphDefault"
      >
        Wrong email or password, please try again.
      </p>
    </div>
  </div>
</div>
`;

exports[`<NotificationBanners /> will render notification banner with minimised class 1`] = `
<div>
  <div
    class="banners"
  >
    <div
      class="root error minimised"
    >
      <p
        class="paragraphDefault"
      >
        Wrong email or password, please try again.
      </p>
    </div>
  </div>
</div>
`;
